### ISSUE TITLE
Linting redundant conditionals?

### ISSUE THREAD

**Issue by IndrajeetPatil** [r-lib/lintr#2197]
I am not sure if this deserves a separate linter, but maybe this example case makes you think of some other cases that could be covered by such a linter.

Arguably, it is hard to write such code intentionally, but it is likely to happen as the outer and inner conditionals start to get distant from each other. 

## Will produce lint

``` r
"if (expr1) {
  # big block of code
  expr2 <- f(expr1)
  if (expr1 && expr2) { # this should lint
    # more code
  }
  # some other code
}" -> code

library(lintr)
lint(text = code, linters = all_linters())
```

<sup>Created on 2023-09-20 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>

## Okay

```r
if (expr1) {
  # big block of code
  expr2 <- f(expr1)
  if (expr2) { # no lint
    # more code
  }
  # some other code
}
```

**Comment by MichaelChirico** [comment]
looks tough to catch most instances, but agree it could lint. no existing linter comes to mind, so filling under new linters.

I encounter this issue most often with case_when statements, which could be included.

**Comment by AshesITR** [comment]
As with the other issue, the challenge will be to correctly detect independence between `expr1` and the first code chunk.
